tests/main/snapd-snap/task.yaml

```diff
@@ -49,19 +42,18 @@ prepare: |
     # shellcheck source=tests/lib/systems.sh
     . "$TESTSLIB/systems.sh"
 
+    if os.query is-fedora || os.query is-arch-linux; then
+        # set up classic execution support
+        ln -sf "$SNAP_MOUNT_DIR" /snap
+        tests.cleanup defer rm -r /snap
```

Comment:
I don't think you want `-r` here, right? Just want to remove the symlink, not contents of `/snap` ?